components/developer/rust/Makefile

```diff
@@ -120,19 +134,18 @@ CONFIGURE_OPTIONS += --enable-rpath
 CONFIGURE_OPTIONS += --disable-codegen-tests
 CONFIGURE_OPTIONS += --disable-dist-src
 CONFIGURE_OPTIONS += --disable-llvm-static-stdcpp
-CONFIGURE_OPTIONS += --enable-ninja
 CONFIGURE_OPTIONS += --enable-vendor
 CONFIGURE_OPTIONS += --disable-docs
 CONFIGURE_OPTIONS += --disable-compiler-docs
 CONFIGURE_OPTIONS += --disable-cargo-native-static
 CONFIGURE_OPTIONS += --enable-option-checking
-CONFIGURE_OPTIONS += --enable-optimize
 CONFIGURE_OPTIONS += --release-channel=stable
 CONFIGURE_OPTIONS += --python=$(PYTHON)
 CONFIGURE_OPTIONS += --enable-llvm-link-shared
 CONFIGURE_OPTIONS += --llvm-root=$(CLANG_ROOT)
-
-COMPONENT_PRE_BUILD_ACTION=	rustup default  $(RUST_PRE_VERSION)-$(RUST_ARCH);
+CONFIGURE_OPTIONS += --enable-llvm-version-check
+CONFIGURE_OPTIONS += --enable-optimize-llvm
+CONFIGURE_OPTIONS += --enable-control-flow-guard
 
 # Cleanup standard environment!
 COMPONENT_BUILD_ENV =
```

Comment:
...and then the `COMPONENT_BUILD_ENV` is cleaned here.

Definitely, this `Makefile` needs some love.